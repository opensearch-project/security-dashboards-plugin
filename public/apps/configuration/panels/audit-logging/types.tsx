/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

export interface GeneralSettings {
  // layer
  enable_rest?: boolean;
  enable_transport?: boolean;
  disabled_rest_categories?: string[];
  disabled_transport_categories?: string[];

  // attribute
  resolve_bulk_requests?: boolean;
  resolve_indices?: boolean;
  log_request_body?: boolean;
  exclude_sensitive_headers?: boolean;

  // ignore
  ignore_users?: string[];
  ignore_requests?: string[];
}

export interface ComplianceSettings {
  enabled?: boolean;

  // read
  read_metadata_only?: boolean;
  read_ignore_users?: string[];
  read_watched_fields?: {
    [key: string]: string;
  };

  // write
  write_metadata_only?: boolean;
  write_log_diffs?: boolean;
  write_ignore_users?: string[];
  write_watched_indices?: string[];
}

export interface AuditLoggingSettings {
  enabled?: boolean;
  audit?: GeneralSettings;
  compliance?: ComplianceSettings;
}
