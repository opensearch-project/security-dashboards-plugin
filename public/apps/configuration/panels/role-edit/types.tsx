/*
 *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { EuiComboBoxOptionOption } from '@elastic/eui';

export type ComboBoxOptions = EuiComboBoxOptionOption[];
export type FieldLevelSecurityMethod = 'exclude' | 'include';
export interface RoleIndexPermissionStateClass {
  indexPatterns: ComboBoxOptions;
  docLevelSecurity: string;
  fieldLevelSecurityMethod: FieldLevelSecurityMethod;
  fieldLevelSecurityFields: ComboBoxOptions;
  maskedFields: ComboBoxOptions;
  allowedActions: ComboBoxOptions;
}

export enum TenantPermissionType {
  None = '',
  Read = 'r',
  Write = 'w',
  Full = 'rw',
}

export interface RoleTenantPermissionStateClass {
  tenantPatterns: ComboBoxOptions;
  permissionType: TenantPermissionType;
}
