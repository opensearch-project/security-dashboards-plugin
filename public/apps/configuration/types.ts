/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { EuiComboBoxOptionOption } from '@elastic/eui';

export type ComboBoxOptions = EuiComboBoxOptionOption[];

export type FieldLevelSecurityMethod = 'exclude' | 'include';

export enum ResourceType {
  roles = 'roles',
  users = 'users',
  permissions = 'permissions',
  tenants = 'tenants',
  tenantsManageTab = 'tenantsManageTab',
  tenantsConfigureTab = 'tenantsConfigureTab',
  auth = 'auth',
  auditLogging = 'auditLogging',
}

export enum Action {
  view = 'view',
  create = 'create',
  edit = 'edit',
  duplicate = 'duplicate',
  manage = 'manage',
  configure = 'configure',
}

export enum SubAction {
  mapuser = 'mapuser',
}

export enum PageId {
  dashboardId = 'dashboards',
  visualizationId = 'visualize',
}

export enum TenantPermissionType {
  None = '',
  Read = 'Read only',
  ReadWrite = 'Read and Write',
}

export interface RouteItem {
  name: string;
  href: string;
}

export interface DataObject<T> {
  [key: string]: T;
}

export interface ObjectsMessage<T> {
  total: number;
  data: DataObject<T>;
}

export interface RoleIndexPermission {
  index_patterns: string[];
  dls: string;
  fls: string[];
  masked_fields: string[];
  allowed_actions: string[];
}

export interface RoleIndexPermissionView extends RoleIndexPermission {
  id: number;
}

export interface RoleTenantPatterns {
  tenant_patterns: string[];
}

export interface RoleTenantPermission extends RoleTenantPatterns {
  allowed_actions: string[];
}

export interface RoleTenantPermissionView extends RoleTenantPatterns {
  permissionType: string;
}

export interface RoleUpdate {
  cluster_permissions: string[];
  index_permissions: RoleIndexPermission[];
  tenant_permissions: RoleTenantPermission[];
}

export interface RoleDetail extends RoleUpdate {
  reserved: boolean;
}

export interface RoleMappingDetail {
  backend_roles: string[];
  hosts: string[];
  users: string[];
}

export interface TenantUpdate {
  description: string;
}

export interface TenantName {
  tenant: string;
}

export interface Tenant extends TenantUpdate, TenantName {
  reserved: boolean;
  tenantValue: string;
}

export interface TenantSelect extends TenantName {
  username: string;
}

export interface RoleTenantPermissionDetail extends RoleTenantPermissionView, Tenant {}

export interface UserAttributes {
  [key: string]: string;
}

export interface InternalUser {
  attributes: UserAttributes;
  backend_roles: string[];
}

export interface InternalUserUpdate extends InternalUser {
  password?: string;
}

export interface ActionGroupUpdate {
  allowed_actions: string[];
}

export interface ActionGroupItem extends ActionGroupUpdate {
  reserved: boolean;
  type: 'cluster' | 'index' | 'all' | 'kibana' | undefined;
}

export interface ExpandedRowMapInterface {
  [key: string]: React.ReactNode;
}

export interface FormRowDeps {
  headerText: string;
  optional?: boolean;
  headerSubText?: string;
  helpLink?: string;
  helpText?: string;
}
