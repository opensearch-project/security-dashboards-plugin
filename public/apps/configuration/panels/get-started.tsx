/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import {
  EuiButton,
  EuiCode,
  EuiFlexGroup,
  EuiFlexItem,
  EuiImage,
  EuiPageHeader,
  EuiPanel,
  EuiSpacer,
  EuiSteps,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import React from 'react';
import { FormattedMessage } from '@osd/i18n/react';
import { flow } from 'lodash';
import { HashRouter as Router, Route } from 'react-router-dom';
import { AppDependencies } from '../../types';
import securityStepsDiagram from '../../../assets/get_started.svg';
import { buildHashUrl, buildUrl } from '../utils/url-builder';
import { Action, ResourceType, RouteItem } from '../types';
import { API_ENDPOINT_CACHE, DocLinks } from '../constants';
import { ExternalLink, ExternalLinkButton } from '../utils/display-utils';
import { TenantList } from './tenant-list/tenant-list';
import { getBreadcrumbs } from '../app-router';

import { CrossPageToast } from '../cross-page-toast';

const addBackendStep = {
  title: 'Add backends',
  children: (
    <>
      <EuiText size="s" color="subdued">
        Add authentication<EuiCode>(authc)</EuiCode>and authorization<EuiCode>(authz)</EuiCode>
        information to<EuiCode>config/opensearch-security/config.yml</EuiCode>. The
        <EuiCode>authc</EuiCode> section contains the backends to check user credentials against.
        The <EuiCode>authz</EuiCode>
        section contains any backends to fetch backend roles from. The most common example of a
        backend role is an LDAP group. <ExternalLink href={DocLinks.AuthenticationFlowDoc} />
      </EuiText>

      <EuiSpacer size="m" />

      <EuiFlexGroup gutterSize="s" wrap>
        <EuiFlexItem grow={false}>
          <ExternalLinkButton
            fill
            href={DocLinks.BackendConfigurationDoc}
            text="Create config.yml"
          />
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiButton
            data-test-subj="review-authentication-and-authorization"
            onClick={() => {
              window.location.href = buildHashUrl(ResourceType.auth);
            }}
          >
            Review authentication and authorization
          </EuiButton>
        </EuiFlexItem>
      </EuiFlexGroup>

      <EuiSpacer size="l" />
    </>
  ),
};

const setOfSteps = [
  {
    title: 'Create roles',
    children: (
      <>
        <EuiText size="s" color="subdued">
          Roles are reusable collections of permissions. The default roles are a great starting
          point, but you might need to create custom roles that meet your exact needs.{' '}
          <ExternalLink href={DocLinks.CreateRolesDoc} />
        </EuiText>

        <EuiSpacer size="m" />

        <EuiFlexGroup gutterSize="s">
          <EuiFlexItem grow={false}>
            <EuiButton
              data-test-subj="explore-existing-roles"
              fill
              onClick={() => {
                window.location.href = buildHashUrl(ResourceType.roles);
              }}
            >
              Explore existing roles
            </EuiButton>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiButton
              data-test-subj="create-new-role"
              onClick={() => {
                window.location.href = buildHashUrl(ResourceType.roles, Action.create);
              }}
            >
              Create new role
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>

        <EuiSpacer size="l" />
      </>
    ),
  },
  {
    title: 'Map users',
    children: (
      <>
        <EuiText size="s" color="subdued">
          After a user successfully authenticates, the security plugin retrieves that userâ€™s roles.
          You can map roles directly to users, but you can also map them to backend roles.{' '}
          <ExternalLink href={DocLinks.MapUsersToRolesDoc} />
        </EuiText>

        <EuiSpacer size="m" />

        <EuiFlexGroup gutterSize="s">
          <EuiFlexItem grow={false}>
            <EuiButton
              data-test-subj="map-users-to-role"
              fill
              onClick={() => {
                window.location.href = buildHashUrl(ResourceType.users);
              }}
            >
              Map users to a role
            </EuiButton>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiButton
              data-test-subj="create-internal-user"
              onClick={() => {
                window.location.href = buildHashUrl(ResourceType.users, Action.create);
              }}
            >
              Create internal user
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
      </>
    ),
  },
];

export function GetStarted(props: AppDependencies) {
  let steps;
  if (props.config.ui.backend_configurable) {
    steps = [addBackendStep, ...setOfSteps];
  } else {
    steps = setOfSteps;
  }

  return (
    <>
      <div className="panel-restrict-width">
        <EuiPageHeader>
          <EuiTitle size="l">
            <h1>Get started</h1>
          </EuiTitle>
          <ExternalLinkButton text="Open in new window" href={buildHashUrl()} />
        </EuiPageHeader>

        <EuiPanel paddingSize="l">
          <EuiText size="s" color="subdued">
            <p>
              The OpenSearch security plugin lets you define the API calls that users can make and
              the data they can access. The most basic configuration consists of these steps.
            </p>
          </EuiText>

          <EuiSpacer size="l" />
          {props.config.ui.backend_configurable && (
            <div className="text-center">
              <EuiImage
                size="xl"
                alt="Three steps to set up your security"
                url={securityStepsDiagram}
                style={{ maxWidth: '100%', height: 'auto', width: 'auto' }}
              />
            </div>
          )}

          <EuiSpacer size="l" />

          <EuiSteps steps={steps} />
        </EuiPanel>

        <EuiSpacer size="l" />

        <EuiPanel paddingSize="l">
          <EuiTitle size="s">
            <h3>Optional: Configure audit logs</h3>
          </EuiTitle>
          <EuiText size="s" color="subdued">
            <p>
              <FormattedMessage
                id="audit.logs.introduction"
                defaultMessage="Audit logs let you track user access to your OpenSearch cluster and are useful for compliance purposes."
              />{' '}
              <ExternalLink href={DocLinks.AuditLogsDoc} />
            </p>
            <EuiButton
              data-test-subj="review-audit-log-configuration"
              fill
              onClick={() => {
                window.location.href = buildHashUrl(ResourceType.auditLogging);
              }}
            >
              Review Audit Log Configuration
            </EuiButton>
          </EuiText>
        </EuiPanel>

        <EuiSpacer size="l" />

        <EuiPanel paddingSize="l">
          <EuiTitle size="s">
            <h3>Optional: Purge cache</h3>
          </EuiTitle>
          <EuiText size="s" color="subdued">
            <p>
              By default, the security plugin caches authenticated users, along with their roles and
              permissions. This option will purge cached users, roles and permissions.
            </p>
            <EuiButton
              iconType="refresh"
              fill
              onClick={() => {
                props.coreStart.http.delete(API_ENDPOINT_CACHE);
              }}
            >
              Purge cache
            </EuiButton>
          </EuiText>
        </EuiPanel>

        <EuiSpacer size="l" />

        <EuiPanel paddingSize="l">
          <EuiTitle size="s">
            <h3>Optional: Multi-tenancy</h3>
          </EuiTitle>
          <EuiText size="s" color="subdued">
            <p>
              By default tenancy is activated in Dashboards. Tenants in OpenSearch Dashboards are
              spaces for saving index patterns, visualizations, dashboards, and other OpenSearch
              Dashboards objects.
            </p>
            <EuiFlexGroup gutterSize="s">
              <EuiFlexItem grow={false}>
                <EuiButton
                  fill
                  onClick={() => {
                    window.location.href = buildHashUrl(ResourceType.tenants);
                  }}
                >
                  Manage Multi-tenancy
                </EuiButton>
              </EuiFlexItem>
              <EuiFlexItem grow={false}>
                <EuiButton
                  onClick={() => {
                    window.location.href = buildHashUrl(ResourceType.tenantsConfigureTab);
                  }}
                >
                  Configure Multi-tenancy
                </EuiButton>
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiText>
        </EuiPanel>
      </div>
    </>
  );
}
