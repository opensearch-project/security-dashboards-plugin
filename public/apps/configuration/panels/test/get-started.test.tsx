/*
 *   Copyright OpenSearch Contributors
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import { shallow } from 'enzyme';
import React from 'react';
import { EuiSteps } from '@elastic/eui';
import { Action } from '../../types';
import { ResourceType } from '../../../../../common';
import { buildHashUrl } from '../../utils/url-builder';
import { GetStarted } from '../get-started';
import * as ToastUtils from '../../utils/toast-utils'; // Import all functions from toast-utils
import * as RequestUtils from '../../utils/request-utils'; // Import all functions from request-utils

jest.mock('../../utils/toast-utils', () => ({
  createSuccessToast: jest.fn(),
  createUnknownErrorToast: jest.fn(),
  useToastState: jest.fn().mockReturnValue([[], jest.fn(), jest.fn()]),
}));

jest.mock('../../utils/request-utils', () => ({
  httpDelete: jest.fn(),
}));

describe('Get started (landing page)', () => {
  const mockCoreStart = {
    http: 1,
  };
  const config = {
    ui: {
      backend_configurable: true,
    },
  };

  it('renders when backend configuration is enabled', () => {
    const component = shallow(
      <GetStarted
        coreStart={mockCoreStart as any}
        navigation={{} as any}
        params={{} as any}
        config={config as any}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders when backend configuration is disabled', () => {
    const config1 = {
      ui: {
        backend_configurable: false,
      },
    };
    const component = shallow(
      <GetStarted
        coreStart={mockCoreStart as any}
        navigation={{} as any}
        params={{} as any}
        config={config1 as any}
      />
    );
    expect(component).toMatchSnapshot();
  });

  describe('should handle assignments to window.location.href correctly', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(
        <GetStarted
          coreStart={mockCoreStart as any}
          navigation={{} as any}
          params={{} as any}
          config={config as any}
        />
      );
      jest.clearAllMocks();
    });

    it('Review authentication and authorization button click', () => {
      const component = wrapper.find(EuiSteps).shallow();
      const button = component.find('[data-test-subj="review-authentication-and-authorization"]');
      expect(button).toHaveLength(1);
      button.simulate('click');
      expect(window.location.hash).toBe(buildHashUrl(ResourceType.auth));
    });

    it('Explore existing roles button click', () => {
      const component = wrapper.find(EuiSteps).shallow();
      const button = component.find('[data-test-subj="explore-existing-roles"]');
      expect(button).toHaveLength(1);
      button.simulate('click');
      expect(window.location.hash).toBe(buildHashUrl(ResourceType.roles));
    });

    it('Create new role button click', () => {
      const component = wrapper.find(EuiSteps).shallow();
      const button = component.find('[data-test-subj="create-new-role"]');
      expect(button).toHaveLength(1);
      button.simulate('click');
      expect(window.location.hash).toBe(buildHashUrl(ResourceType.roles, Action.create));
    });

    it('Map users to a role button click', () => {
      const component = wrapper.find(EuiSteps).shallow();
      const button = component.find('[data-test-subj="map-users-to-role"]');
      expect(button).toHaveLength(1);
      button.simulate('click');
      expect(window.location.hash).toBe(buildHashUrl(ResourceType.users));
    });

    it('Create internal user button click', () => {
      const component = wrapper.find(EuiSteps).shallow();
      const button = component.find('[data-test-subj="create-internal-user"]');
      expect(button).toHaveLength(1);
      button.simulate('click');
      expect(window.location.hash).toBe(buildHashUrl(ResourceType.users, Action.create));
    });

    it('Review Audit Log Configuration button click', () => {
      const button = wrapper.find('[data-test-subj="review-audit-log-configuration"]');
      expect(button).toHaveLength(1);
      button.simulate('click');
      expect(window.location.hash).toBe(buildHashUrl(ResourceType.auditLogging));
    });
  });

  describe('Tests purge cache button', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(
        <GetStarted
          coreStart={mockCoreStart as any}
          navigation={{} as any}
          params={{} as any}
          config={config as any}
        />
      );
      jest.clearAllMocks();
    });

    it('Purge cache button fails', async () => {
      const button = wrapper.find('[data-test-subj="purge-cache"]');
      expect(button).toHaveLength(1);

      // Failure case: Mock httpDelete to reject
      jest
        .spyOn(RequestUtils, 'httpDelete')
        .mockRejectedValueOnce(new Error('Failed to purge cache'));

      await button.props().onClick(); // Simulate button click
      expect(ToastUtils.createUnknownErrorToast).toHaveBeenCalledTimes(1);
    });

    it('Purge cache button works', async () => {
      const button = wrapper.find('[data-test-subj="purge-cache"]');
      expect(button).toHaveLength(1);

      // Success case: Mock httpDelete to resolve
      jest.spyOn(RequestUtils, 'httpDelete').mockResolvedValueOnce('nice');
      await button.props().onClick(); // Simulate button click
      expect(ToastUtils.createSuccessToast).toHaveBeenCalledTimes(1);
    });
  });
});
