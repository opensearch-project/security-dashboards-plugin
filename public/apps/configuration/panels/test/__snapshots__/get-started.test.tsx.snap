// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get started (landing page) renders when backend configuration is disabled 1`] = `
<Fragment>
  <div
    className="panel-restrict-width"
  >
    <EuiPageHeader>
      <EuiTitle
        size="l"
      >
        <h1>
          Get started
        </h1>
      </EuiTitle>
      <ExternalLinkButton
        href="#/"
        text="Open in new window"
      />
    </EuiPageHeader>
    <EuiPanel
      paddingSize="l"
    >
      <EuiText
        color="subdued"
        size="s"
      >
        <p>
          The OpenSearch security plugin lets you define the API calls that users can make and the data they can access. The most basic configuration consists of these steps.
        </p>
      </EuiText>
      <EuiSpacer
        size="l"
      />
      <EuiSpacer
        size="l"
      />
      <EuiSteps
        steps={
          Array [
            Object {
              "children": <React.Fragment>
                <EuiText
                  color="subdued"
                  size="s"
                >
                  Roles are reusable collections of permissions. The default roles are a great starting point, but you might need to create custom roles that meet your exact needs.
                   
                  <ExternalLink
                    href="https://opensearch.org/docs/latest/security-plugin/access-control/users-roles/#create-roles"
                  />
                </EuiText>
                <EuiSpacer
                  size="m"
                />
                <EuiFlexGroup
                  gutterSize="s"
                >
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiButton
                      data-test-subj="explore-existing-roles"
                      fill={true}
                      onClick={[Function]}
                    >
                      Explore existing roles
                    </EuiButton>
                  </EuiFlexItem>
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiButton
                      data-test-subj="create-new-role"
                      onClick={[Function]}
                    >
                      Create new role
                    </EuiButton>
                  </EuiFlexItem>
                </EuiFlexGroup>
                <EuiSpacer
                  size="l"
                />
              </React.Fragment>,
              "title": "Create roles",
            },
            Object {
              "children": <React.Fragment>
                <EuiText
                  color="subdued"
                  size="s"
                >
                  After a user successfully authenticates, the security plugin retrieves that userâ€™s roles. You can map roles directly to users, but you can also map them to backend roles.
                   
                  <ExternalLink
                    href="https://opensearch.org/docs/latest/security-plugin/access-control/users-roles/#map-users-to-roles"
                  />
                </EuiText>
                <EuiSpacer
                  size="m"
                />
                <EuiFlexGroup
                  gutterSize="s"
                >
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiButton
                      data-test-subj="map-users-to-role"
                      fill={true}
                      onClick={[Function]}
                    >
                      Map users to a role
                    </EuiButton>
                  </EuiFlexItem>
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiButton
                      data-test-subj="create-internal-user"
                      onClick={[Function]}
                    >
                      Create internal user
                    </EuiButton>
                  </EuiFlexItem>
                </EuiFlexGroup>
              </React.Fragment>,
              "title": "Map users",
            },
          ]
        }
      />
    </EuiPanel>
    <EuiSpacer
      size="l"
    />
    <EuiPanel
      paddingSize="l"
    >
      <EuiTitle
        size="s"
      >
        <h3>
          Optional: Configure audit logs
        </h3>
      </EuiTitle>
      <EuiText
        color="subdued"
        size="s"
      >
        <p>
          <FormattedMessage
            defaultMessage="Audit logs let you track user access to your OpenSearch cluster and are useful for compliance purposes."
            id="audit.logs.introduction"
            values={Object {}}
          />
           
          <ExternalLink
            href="https://opensearch.org/docs/latest/security-plugin/audit-logs/"
          />
        </p>
        <EuiButton
          data-test-subj="review-audit-log-configuration"
          fill={true}
          onClick={[Function]}
        >
          Review Audit Log Configuration
        </EuiButton>
      </EuiText>
    </EuiPanel>
    <EuiSpacer
      size="l"
    />
    <EuiPanel
      paddingSize="l"
    >
      <EuiTitle
        size="s"
      >
        <h3>
          Optional: Purge cache
        </h3>
      </EuiTitle>
      <EuiText
        color="subdued"
        size="s"
      >
        <p>
          By default, the security plugin caches authenticated users, along with their roles and permissions. This option will purge cached users, roles and permissions.
        </p>
        <EuiButton
          data-test-subj="purge-cache"
          fill={true}
          iconType="refresh"
          onClick={[Function]}
        >
          Purge cache
        </EuiButton>
      </EuiText>
    </EuiPanel>
    <EuiSpacer
      size="l"
    />
    <EuiPanel
      paddingSize="l"
    >
      <EuiTitle
        size="s"
      >
        <h3>
          Optional: Multi-tenancy
        </h3>
      </EuiTitle>
      <EuiText
        color="subdued"
        size="s"
      >
        <p>
          By default tenancy is activated in Dashboards. Tenants in OpenSearch Dashboards are spaces for saving index patterns, visualizations, dashboards, and other OpenSearch Dashboards objects.
        </p>
        <EuiFlexGroup
          gutterSize="s"
        >
          <EuiFlexItem
            grow={false}
          >
            <EuiButton
              fill={true}
              onClick={[Function]}
            >
              Manage Multi-tenancy
            </EuiButton>
          </EuiFlexItem>
          <EuiFlexItem
            grow={false}
          >
            <EuiButton
              onClick={[Function]}
            >
              Configure Multi-tenancy
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiText>
    </EuiPanel>
  </div>
  <EuiGlobalToastList
    dismissToast={[MockFunction]}
    side="right"
    toastLifeTimeMs={10000}
    toasts={Array []}
  />
</Fragment>
`;

exports[`Get started (landing page) renders when backend configuration is enabled 1`] = `
<Fragment>
  <div
    className="panel-restrict-width"
  >
    <EuiPageHeader>
      <EuiTitle
        size="l"
      >
        <h1>
          Get started
        </h1>
      </EuiTitle>
      <ExternalLinkButton
        href="#/"
        text="Open in new window"
      />
    </EuiPageHeader>
    <EuiPanel
      paddingSize="l"
    >
      <EuiText
        color="subdued"
        size="s"
      >
        <p>
          The OpenSearch security plugin lets you define the API calls that users can make and the data they can access. The most basic configuration consists of these steps.
        </p>
      </EuiText>
      <EuiSpacer
        size="l"
      />
      <div
        className="security-steps-diagram"
      />
      <EuiSpacer
        size="l"
      />
      <EuiSteps
        steps={
          Array [
            Object {
              "children": <React.Fragment>
                <EuiText
                  color="subdued"
                  size="s"
                >
                  Add authentication
                  <EuiCode>
                    (authc)
                  </EuiCode>
                  and authorization
                  <EuiCode>
                    (authz)
                  </EuiCode>
                  information to
                  <EuiCode>
                    config/opensearch-security/config.yml
                  </EuiCode>
                  . The
                  <EuiCode>
                    authc
                  </EuiCode>
                   section contains the backends to check user credentials against. The 
                  <EuiCode>
                    authz
                  </EuiCode>
                  section contains any backends to fetch backend roles from. The most common example of a backend role is an LDAP group. 
                  <ExternalLink
                    href="https://opensearch.org/docs/latest/security-plugin/configuration/concepts/"
                  />
                </EuiText>
                <EuiSpacer
                  size="m"
                />
                <EuiFlexGroup
                  gutterSize="s"
                  wrap={true}
                >
                  <EuiFlexItem
                    grow={false}
                  >
                    <ExternalLinkButton
                      fill={true}
                      href="https://opensearch.org/docs/latest/security-plugin/configuration/configuration/"
                      text="Create config.yml"
                    />
                  </EuiFlexItem>
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiButton
                      data-test-subj="review-authentication-and-authorization"
                      onClick={[Function]}
                    >
                      Review authentication and authorization
                    </EuiButton>
                  </EuiFlexItem>
                </EuiFlexGroup>
                <EuiSpacer
                  size="l"
                />
              </React.Fragment>,
              "title": "Add backends",
            },
            Object {
              "children": <React.Fragment>
                <EuiText
                  color="subdued"
                  size="s"
                >
                  Roles are reusable collections of permissions. The default roles are a great starting point, but you might need to create custom roles that meet your exact needs.
                   
                  <ExternalLink
                    href="https://opensearch.org/docs/latest/security-plugin/access-control/users-roles/#create-roles"
                  />
                </EuiText>
                <EuiSpacer
                  size="m"
                />
                <EuiFlexGroup
                  gutterSize="s"
                >
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiButton
                      data-test-subj="explore-existing-roles"
                      fill={true}
                      onClick={[Function]}
                    >
                      Explore existing roles
                    </EuiButton>
                  </EuiFlexItem>
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiButton
                      data-test-subj="create-new-role"
                      onClick={[Function]}
                    >
                      Create new role
                    </EuiButton>
                  </EuiFlexItem>
                </EuiFlexGroup>
                <EuiSpacer
                  size="l"
                />
              </React.Fragment>,
              "title": "Create roles",
            },
            Object {
              "children": <React.Fragment>
                <EuiText
                  color="subdued"
                  size="s"
                >
                  After a user successfully authenticates, the security plugin retrieves that userâ€™s roles. You can map roles directly to users, but you can also map them to backend roles.
                   
                  <ExternalLink
                    href="https://opensearch.org/docs/latest/security-plugin/access-control/users-roles/#map-users-to-roles"
                  />
                </EuiText>
                <EuiSpacer
                  size="m"
                />
                <EuiFlexGroup
                  gutterSize="s"
                >
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiButton
                      data-test-subj="map-users-to-role"
                      fill={true}
                      onClick={[Function]}
                    >
                      Map users to a role
                    </EuiButton>
                  </EuiFlexItem>
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiButton
                      data-test-subj="create-internal-user"
                      onClick={[Function]}
                    >
                      Create internal user
                    </EuiButton>
                  </EuiFlexItem>
                </EuiFlexGroup>
              </React.Fragment>,
              "title": "Map users",
            },
          ]
        }
      />
    </EuiPanel>
    <EuiSpacer
      size="l"
    />
    <EuiPanel
      paddingSize="l"
    >
      <EuiTitle
        size="s"
      >
        <h3>
          Optional: Configure audit logs
        </h3>
      </EuiTitle>
      <EuiText
        color="subdued"
        size="s"
      >
        <p>
          <FormattedMessage
            defaultMessage="Audit logs let you track user access to your OpenSearch cluster and are useful for compliance purposes."
            id="audit.logs.introduction"
            values={Object {}}
          />
           
          <ExternalLink
            href="https://opensearch.org/docs/latest/security-plugin/audit-logs/"
          />
        </p>
        <EuiButton
          data-test-subj="review-audit-log-configuration"
          fill={true}
          onClick={[Function]}
        >
          Review Audit Log Configuration
        </EuiButton>
      </EuiText>
    </EuiPanel>
    <EuiSpacer
      size="l"
    />
    <EuiPanel
      paddingSize="l"
    >
      <EuiTitle
        size="s"
      >
        <h3>
          Optional: Purge cache
        </h3>
      </EuiTitle>
      <EuiText
        color="subdued"
        size="s"
      >
        <p>
          By default, the security plugin caches authenticated users, along with their roles and permissions. This option will purge cached users, roles and permissions.
        </p>
        <EuiButton
          data-test-subj="purge-cache"
          fill={true}
          iconType="refresh"
          onClick={[Function]}
        >
          Purge cache
        </EuiButton>
      </EuiText>
    </EuiPanel>
    <EuiSpacer
      size="l"
    />
    <EuiPanel
      paddingSize="l"
    >
      <EuiTitle
        size="s"
      >
        <h3>
          Optional: Multi-tenancy
        </h3>
      </EuiTitle>
      <EuiText
        color="subdued"
        size="s"
      >
        <p>
          By default tenancy is activated in Dashboards. Tenants in OpenSearch Dashboards are spaces for saving index patterns, visualizations, dashboards, and other OpenSearch Dashboards objects.
        </p>
        <EuiFlexGroup
          gutterSize="s"
        >
          <EuiFlexItem
            grow={false}
          >
            <EuiButton
              fill={true}
              onClick={[Function]}
            >
              Manage Multi-tenancy
            </EuiButton>
          </EuiFlexItem>
          <EuiFlexItem
            grow={false}
          >
            <EuiButton
              onClick={[Function]}
            >
              Configure Multi-tenancy
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiText>
    </EuiPanel>
  </div>
  <EuiGlobalToastList
    dismissToast={[MockFunction]}
    side="right"
    toastLifeTimeMs={10000}
    toasts={Array []}
  />
</Fragment>
`;
